### PR TITLE
add network_chaos.md

### PR DESCRIPTION
**Pull request by fewdan**
Add docs for network-chaos.  Explain the purpose of each parameter in yaml.

### CODE REVIEW

File: doc/network_chaos.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+* **selector** is used to select pods that are used to inject chaos action.
+* **direction** represents the partition direction, supported direction: from / to / both.
+* **target** represents network partition target.
+* **duration** define the duration time for each chaos experiment. As the example shows, the network partition lasts 10 seconds.
+* **scheduler** defines some scheduler rules to the running time of the chaos experiment about pods. More rule info: https://godoc.org/github.com/robfig/cron
+
+
+# Network Loss / Delay / Duplicate / Corrupt
+
+> There are 4 cases, loss, delay, duplicate and corrupt.
```
Comment (mapleFU) [inline doc/network_chaos.md]: how about removing this level's `>`?

Comment (fewdan) [reply doc/network_chaos.md]: ok. I have removed.

File: doc/network_chaos.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    selector:
+      namespaces:
+        - tidb-cluster-demo
+      labelSelectors:
+        "app.kubernetes.io/component": "tikv"
+    mode: one
+  duration: "10s"
+  scheduler:
+    cron: "@every 15s"
+```
```
Comment (cwen0) [inline doc/network_chaos.md]: add example file addr 
eg:  
```
User can find and edit the template refer to examples/io-mixed-sample.yaml.
```

File: doc/network_chaos.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+This document whill help user to build Network Chaos experiments.
+
+Network chaos are mainly divided into two categories, namely **netem chaos** and **network partition**.
+
+Netem chaos contains some kinds of network chaos, such as delay, duplication, loss and corrupt.
+
+Network partition can decompose pods into several independent subnets by blocking communication between them.
+
+
+# Network Partition Action
```
Comment (mahjonp) [inline doc/network_chaos.md]:
```suggestion
# Network Partition Action
## Network Partition Action
```

File: doc/network_chaos.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+* **action** defines the specific pod chaos action. In this case, it means network partition, represents the chaos action of network partition of pods.
+* **mode** defines the mode to run chaos action.
+* **selector** is used to select pods that are used to inject chaos action.
+* **direction** represents the partition direction, supported direction: from / to / both.
+* **target** represents network partition target.
+* **duration** define the duration time for each chaos experiment. As the example shows, the network partition lasts 10 seconds.
+* **scheduler** defines some scheduler rules to the running time of the chaos experiment about pods. More rule info: https://godoc.org/github.com/robfig/cron
+
+
+# Netem Chaos Actions
```
Comment (mahjonp) [inline doc/network_chaos.md]:
```suggestion
# Netem Chaos Actions
## Netem Chaos Actions
```

Comment (fewdan) [reply doc/network_chaos.md]: Directory at this level has been modified.

File: doc/network_chaos.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+### Network Loss
+Network Loss means that network packets are dropped randomly.
+> In this case, two attributes are required, loss and correlation.
+>
+> ```yaml
+> loss:
+>   loss: "25"
+>   correlation: "25"
+> ```
+> Loss represents the percentage of packet loss. The above example shows a 25% chance of parcket loss.
```
Comment (mahjonp) [inline doc/network_chaos.md]: s/parcket/packet/g

File: doc/network_chaos.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+
+## Netem Chaos Actions
+
+There are 4 cases, loss, delay, duplicate and corrupt.
+
+The meanings of action, mode, selector duration, scheduler are consistent with the description in the Network Partition.
+
+### Network Loss
+Network Loss means that network packets are dropped randomly.
```
Comment (mahjonp) [inline doc/network_chaos.md]: need a blank line

File: doc/network_chaos.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+> loss:
+>   loss: "25"
+>   correlation: "25"
+> ```
+> Loss represents the percentage of packet loss. The above example shows a 25% chance of parcket loss.
+>
+> Network chaos variation isn't purely random, so to emulate that there is a correlation value as well.
+
+### Network Delay
+Network Delay means to delay the sending of network messages.
```
Comment (mahjonp) [inline doc/network_chaos.md]: ditto

File: doc/network_chaos.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+>    jitter: "90ms"
+>```
+> Latency indicates the delay time in sending packets.
+>
+> jitter represents the jitter of the delay time.
+>
+> The above example shows that the network latency is 90ms ± 90ms.
+
+### Network Duplicate
+Network duplicate means packet duplication.
```
Comment (mahjonp) [inline doc/network_chaos.md]: ditto

File: doc/network_chaos.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+>```yaml
+>  duplicate:
+>    duplicate: "40"
+>    correlation: "25"
+>```
+>
+> Network duplicate is specified the same way as network loss. The parameter "Duplicate" indicates the percentage of packet duplication. And it shows that duplication rate is 40%. 
+
+### Network Corrupt
+Network corrupt means packet corruption.
```
Comment (mahjonp) [inline doc/network_chaos.md]: ditto

Comment (fewdan) [reply doc/network_chaos.md]: done

**Comment by zhouqiang-cl** [comment]
@dcalvin PTAL

File: doc/network_chaos.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+> In this case, two attributes are required, loss and correlation.
+>
+> ```yaml
+> loss:
+>   loss: "25"
+>   correlation: "25"
+> ```
+> Loss represents the percentage of packet loss. The above example shows a 25% chance of packet loss.
+>
+> Network chaos variation isn't purely random, so to emulate that there is a correlation value as well.
```
Comment (dcalvin) [inline doc/network_chaos.md]: Can you provide a direct definition for correlation?

File: doc/network_chaos.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+### Network Delay
+
+Network Delay means to delay the sending of network messages.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
A Network Delay action causes delays in message sending. To add a Network Delay action, locate and edit the corresponding template in [/examples](../examples/network-delay-example.yaml).
```

File: doc/network_chaos.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+> In this case, three attributes are required, correlation, jitter and latency.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
> In this case, three action specific attributes are required - correlation, jitter, and latency.
```

File: doc/network_chaos.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+>
+>```yaml
+>  delay:
+>    latency: "90ms"
+>    correlation: "25"
+>    jitter: "90ms"
+>```
+> Latency indicates the delay time in sending packets.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion

> **latency** defines the delay time in sending packets.
```

File: doc/network_chaos.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+>
+> jitter represents the jitter of the delay time.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
> **jitter** specifies the jitter of the delay time.
```

File: doc/network_chaos.md [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+>
+> The above example shows that the network latency is 90ms ± 90ms.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
> In the above example, the network latency is 90ms ± 90ms.
```

File: doc/network_chaos.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+### Network Duplicate
+
+Network duplicate means packet duplication.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
A Network Duplicate action causes packet duplication. To add a Network Duplicate action, locate and edit the corresponding template in [/examples](../examples/network-duplicate-example.yaml).
```

File: doc/network_chaos.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+> In this case, two attributes are required, correlation and duplicate.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
> In this case, two attributes are required - correlation and duplicate.
```

File: doc/network_chaos.md [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+>
+>```yaml
+>  duplicate:
+>    duplicate: "40"
+>    correlation: "25"
+>```
+>
+> Network duplicate is specified the same way as network loss. The parameter "Duplicate" indicates the percentage of packet duplication. And it shows that duplication rate is 40%.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
>   **duplicate** indicates the percentage of packet duplication. In the above example, the duplication rate is 40%. 
```

File: doc/network_chaos.md [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+>  duplicate:
+>    duplicate: "40"
+>    correlation: "25"
+>```
+>
+> Network duplicate is specified the same way as network loss. The parameter "Duplicate" indicates the percentage of packet duplication. And it shows that duplication rate is 40%. 
+
+### Network Corrupt
+
+Network corrupt means packet corruption.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
A Network Corrupt action causes packet corruption. To add a Network Corrupt action, locate and edit the corresponding template in [/examples](../examples/network-corrupt-example.yaml).
```

File: doc/network_chaos.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+> In this case, two attributes are required, correlation and corrupt.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
> In this case, two action specific attributes are required - correlation and corrupt.
```

File: doc/network_chaos.md [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+>
+>```yaml
+>  corrupt:
+>    corrupt: "40"
+>    correlation: "25"
+>```
+>
+> Similar to the other cases described above, the parameter "corrupt" indicates the percentage of packet corruption.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
> **corrupt** specifies the percentage of packet corruption. 
```

File: doc/network_chaos.md [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+A Network Duplicate action causes packet duplication. To add a Network Duplicate action, locate and edit the corresponding template in [/examples](../examples/network-duplicate-example.yaml).
+> In this case, two attributes are required - correlation and duplicate.
+>
+>```yaml
+>  duplicate:
+>    duplicate: "40"
+>    correlation: "25"
+>```
+>
+>   **duplicate** indicates the percentage of packet duplication. In the above example, the duplication rate is 40%.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
>**duplicate** indicates the percentage of packet duplication. In the above example, the duplication rate is 40%. 
```

File: doc/network_chaos.md [hunk 23]
```diff
@@ -0,0 +1,115 @@
+# Network Chaos Document
+
+This document describes how to add network chaos experiments in Chaos Mesh.
+
+Network chaos is mainly divided into the following two categories:
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
Network Chaos actions are mainly divided into the following two categories:
```

File: doc/network_chaos.md [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+
+- **netem chaos** covers regular network faults, such as fault injections into network delay, duplication, loss, and corrupt.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
- **Network Partition** action separates pods into several independent subnets by blocking communication between them.
- **Network Emulation (Netem) Chaos** actions cover regular network faults, such as network delay, duplication, loss, and corruption.
```

File: doc/network_chaos.md [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+
+- **network partition** separates pods into several independent subnets by blocking communication between them.
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
- **Network Partition** action separates pods into several independent subnets by blocking communication between them.
```

File: doc/network_chaos.md [hunk 26]
```diff
@@ -0,0 +1,115 @@
...
+
+## Network Partition Action
+
+To add the network partition chaos experiment, locate and edit the corresponding template in [/examples](./examples/network-partition-example.yaml). Below is a sample network partition configuration file:
```
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
To add the Network Partition chaos experiment, locate and edit the corresponding template in [/examples](./examples/network-partition-example.yaml). Below is a sample network partition configuration file:
```

File: doc/network_chaos.md (hunk 24 above)
Comment (dcalvin) [inline doc/network_chaos.md]:
```suggestion
```

**Pull request merged**